src/boat_simulator/boat_simulator/nodes/low_level_control/controller.py

```diff
@@ -0,0 +1,259 @@
+
+        """
+
+        self.desired_heading = new_desired_heading
+        self.current_heading = new_current_heading
+        self.compute_setpoint()
+
+    def change_desired_heading(self, changed_desired_heading) -> None:
+        """Changes desired heading to a new angle. Used for testing purposes
+
+        Args:
+            `changed_desired_heading` (Scalar): New desired heading in degrees
+
+        """
+        self.desired_heading = changed_desired_heading
+
+
+class SailController(ActuatorController):
+    """General Class for the Actuator Controller.
+
+    SailController Extends: ActuatorController
+    """
+
+    def __init__(
+        self,
+        target_angle: Scalar,
+        current_control_ang: Scalar,
+        time_step: Scalar,
+        kp: Scalar,
+        cp: Scalar,
+        control_speed: Scalar,
+        max_angle_range=SAIL_MAX_ANGLE_RANGE,
+    ):
+        """Initializes the class attributes.
+
+        Args:
+            `target_angle` (Scalar): Target angle for sail controller in degrees.
+             0 degrees (North) at the positive y axis and increasing clockwise.
+            `current_control_ang` (Scalar): Current control mechanism angle in degrees.
+            `time_step` (Scalar): Time per iteration given in seconds.
+            `kp` (Scalar): Proportional constant when calculating error.
+            `cp` (Scalar): Tuning parameter for control action.
+            `control_speed` (Scalar): Speed in which the controller turns in degrees / seconds.
+            'running_error' (Scalar): Error between current and target control angle.
+
+        """
+        super().__init__(
+            current_control_ang,
+            time_step,
+            control_speed,
+            max_angle_range,
+        )
+
+        self.max_angle_range = SAIL_MAX_ANGLE_RANGE
+        self.running_error = 0.0
+        self.target_angle = target_angle
+        self.kp = kp
+        self.cp = cp
+
+    def compute_error(self) -> Scalar:
+        """Computes the corresponding control error angle between current control angle and
+        target control angle
+
+        Returns:
+            Scalar: Corresponding error angle between the
+            current and target control angle in degrees"""
+
+        self.running_error = self.target_angle - self.current_control_ang
+        return self.running_error
+
+    def reset_setpoint(self, new_target: Scalar) -> None:
+        """Resets a new desired sail actuator angle and updates the running_error
+
+        Args:
+            `target_angle` (Scalar): New desired sail controller angle
+
+        """
+        self.target_angle = new_target
+        self.compute_error()
+
+    def reset_target_angle(self, changed_target_angle) -> None:
+        """Changes target_angle desired to a new angle. Used for testing purposes
+
+        Args:
+            `changed_target_angle` (Scalar): New desired heading in degrees
+
+        """
+        self.target_angle = changed_target_angle
```

Comment:
What is the difference between this function and the one above? What is this needed for?